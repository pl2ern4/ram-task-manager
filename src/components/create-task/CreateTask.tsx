import Editor from "../editor/Editor";
import SelectInput from "../select-input/SelectInput";
import {TICKET_STATUS, TICKET_TYPE, USER} from "../../constants";
import Button from '@mui/material/Button';
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import { useGlobalContext } from "@/context/application";
import React, { useState } from "react";
import {
    CustomForm,
    CreateTicketActionButtonGroup,
    CustomInput,
    ErrorDialog,
    DateWrapper
} from './styles';

const CreateTask = ()=>{
    const {setOpenCreateTicket, createTicket } = useGlobalContext();
    const [description, setdescription ] = useState();
    const [error, setError] = useState('');
    const [deadlineDate, setDeadLineDate] = useState();
    const saveData = (event) => {
        const elements = new FormData(event.currentTarget);
        const title = elements.get("title");
        const status = elements.get("status");
        const asignee = elements.get("asignee");
        const taskType = elements.get("taskType");
        const storyPoint = elements.get("storyPoint");
        const reportee = elements.get("reportee");
        if(!title){
            setError("Please enter Correct Title");
            return;
        }
        setError('');
        event.preventDefault();
        createTicket({
            title,
            storyPoint,
            taskType,
            asignee,
            reportee,
            status,
            description,
            deadlineDate
        });
    }

    return (
        <form method="POST" onSubmit={saveData}>
            <CustomForm>
                {error && <ErrorDialog>{error}</ErrorDialog>}
                    <CustomInput name="title" id="title" aria-describedby="my-helper-text" type="text" placeholder="Title" required/>
                    <Editor setDescription={setdescription} label="Description" shouldDisplayBorderOnReadOnly={true} shouldAlwaysEditable={true}/>
                    <SelectInput name="status" label='status' defaultOption='Todo' options={TICKET_STATUS}/>
                    <SelectInput name="asignee" label='Asignee' defaultOption='Not Assigned' options={USER}/>
                    <SelectInput name="reportee" label='Reportee' defaultOption='Not Assigned' options={USER}/>
                    <SelectInput name="taskType" label='Task Type' defaultOption='Not Assigned' options={TICKET_TYPE}/>
                    <CustomInput name="storyPoint" id="storyPoint" aria-describedby="my-helper-text" type="number" placeholder="Story Point"/>
                    <DateWrapper>
                        <span> Deadline of the Ticket? </span>
                        <DatePicker selected={deadlineDate} onChange={setDeadLineDate}/>
                    </DateWrapper>
                    <CreateTicketActionButtonGroup>
                        <Button variant="contained" type="submit">Create Ticket</Button>
                        <Button variant="outlined" onClick={()=>setOpenCreateTicket(false)}>Cancel</Button>
                    </CreateTicketActionButtonGroup>
                
            </CustomForm>
        </form>
    )
}

export default CreateTask;