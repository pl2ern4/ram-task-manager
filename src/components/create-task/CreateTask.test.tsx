import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import CreateTask from './CreateTask';

describe('CreateTask', () => {
    it('should Match Snapshot', async () => {
        const { container } = render(<CreateTask/>)
        expect(container).toMatchSnapshot();
    });
})