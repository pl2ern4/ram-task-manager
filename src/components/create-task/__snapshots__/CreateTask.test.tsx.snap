// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask should Match Snapshot 1`] = `
<div>
  <form
    method="POST"
  >
    <div
      class="MuiFormGroup-root css-1nnfls1-MuiFormGroup-root"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1e7wo54-MuiInputBase-root-MuiInput-root"
        required=""
      >
        <input
          aria-describedby="my-helper-text"
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="title"
          name="title"
          placeholder="Title"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label>
          <h3>
            Description
             :
          </h3>
        </label>
        <div />
      </div>
      <div
        class="css-1dw64bn"
      >
        <label
          class="css-14pxzl7"
        >
          status
        </label>
        <button
          aria-controls=":r0:"
          aria-expanded="false"
          class="base-Select-root css-12xgkxt"
          role="combobox"
          type="button"
        >
          Todo
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UnfoldMoreRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
            />
          </svg>
        </button>
        <input
          aria-hidden="true"
          name="status"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
          tabindex="-1"
          value="Todo"
        />
      </div>
      <div
        class="css-1dw64bn"
      >
        <label
          class="css-14pxzl7"
        >
          Asignee
        </label>
        <button
          aria-controls=":r1:"
          aria-expanded="false"
          class="base-Select-root css-12xgkxt"
          role="combobox"
          type="button"
        >
          Not Assigned
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UnfoldMoreRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
            />
          </svg>
        </button>
        <input
          aria-hidden="true"
          name="asignee"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
          tabindex="-1"
          value="Not Assigned"
        />
      </div>
      <div
        class="css-1dw64bn"
      >
        <label
          class="css-14pxzl7"
        >
          Reportee
        </label>
        <button
          aria-controls=":r2:"
          aria-expanded="false"
          class="base-Select-root css-12xgkxt"
          role="combobox"
          type="button"
        >
          Not Assigned
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UnfoldMoreRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
            />
          </svg>
        </button>
        <input
          aria-hidden="true"
          name="reportee"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
          tabindex="-1"
          value="Not Assigned"
        />
      </div>
      <div
        class="css-1dw64bn"
      >
        <label
          class="css-14pxzl7"
        >
          Task Type
        </label>
        <button
          aria-controls=":r3:"
          aria-expanded="false"
          class="base-Select-root css-12xgkxt"
          role="combobox"
          type="button"
        >
          Not Assigned
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="UnfoldMoreRoundedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
            />
          </svg>
        </button>
        <input
          aria-hidden="true"
          name="taskType"
          style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
          tabindex="-1"
          value="Not Assigned"
        />
      </div>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-1e7wo54-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-describedby="my-helper-text"
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          id="storyPoint"
          name="storyPoint"
          placeholder="Story Point"
          type="number"
          value=""
        />
      </div>
      <div
        class="css-1kq4js9"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Create Ticket
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
